app/src/org/commcare/recovery/measures/ExecuteRecoveryMeasuresPresenter.java

```diff
@@ -138,53 +143,54 @@ private int executeMeasure() {
             return STATUS_FAILED;
         }
 
-        try {
-            switch (mCurrentMeasure.getType()) {
-                case MEASURE_TYPE_APP_REINSTALL_AND_UPDATE:
-                    if (AppUtils.notOnLatestAppVersion()) {
-                        showInstallMethodChooser();
-                        return STATUS_WAITING;
-                    } else {
-                        return STATUS_EXECUTED;
-                    }
-                case MEASURE_TYPE_APP_OFFLINE_REINSTALL_AND_UPDATE:
-                    if (AppUtils.notOnLatestAppVersion()) {
-                        initateAutoCczScan();
-                        return STATUS_WAITING;
-                    } else {
-                        return STATUS_EXECUTED;
-                    }
-                case MEASURE_TYPE_APP_UPDATE:
-                    if (AppUtils.notOnLatestAppVersion()) {
-                        executeAutoUpdate();
-                        return STATUS_WAITING;
-                    } else {
-                        return STATUS_EXECUTED;
-                    }
-                case MEASURE_TYPE_CC_REINSTALL:
-                    if (AppUtils.notOnLatestCCVersion()) {
-                        launchActivity(PromptCCReinstallActivity.class, ExecuteRecoveryMeasuresActivity.PROMPT_APK_REINSTALL);
-                        return STATUS_WAITING;
-                    } else {
-                        return STATUS_EXECUTED;
-                    }
-                case MEASURE_TYPE_CC_UPDATE:
-                    if (AppUtils.notOnLatestCCVersion()) {
-                        launchActivity(PromptApkUpdateActivity.class, ExecuteRecoveryMeasuresActivity.PROMPT_APK_UPDATE);
-                        return STATUS_WAITING;
-                    } else {
-                        return STATUS_EXECUTED;
-                    }
-                default:
-                    // A type that we do not recognize, mark it as executed so that it gets ignored
+
+        switch (mCurrentMeasure.getType()) {
+            case MEASURE_TYPE_APP_REINSTALL_AND_UPDATE:
+                if (AppUtils.notOnLatestAppVersion()) {
+                    showInstallMethodChooser();
+                    return STATUS_WAITING;
+                } else {
                     return STATUS_EXECUTED;
-            }
-        } catch (Exception e) {
```

Comment:
This is only change in L140 - L190. Moved this exception handling to the top level function `start`